bundles/org.openhab.binding.opengarage/src/main/java/org/openhab/binding/opengarage/internal/OpenGarageHandler.java

```diff
@@ -91,78 +100,80 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-            pollStatus();
+            pollAgainIn = pollStatus();
         } catch (IOException e) {
-            logger.debug("Could not connect to OpenGarage controller", e);
+            logger.warn("Could not connect to OpenGarage controller", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
         } catch (RuntimeException e) {
             logger.warn("Unexpected error connecting to OpenGarage controller", e);
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
         }
+        return pollAgainIn;
     }
 
-    private void stopPoll() {
-        final Future<?> future = pollFuture;
-        if (future != null && !future.isCancelled()) {
-            future.cancel(true);
-            pollFuture = null;
-        }
-    }
-
-    private void pollStatus() throws IOException {
+    private long pollStatus() throws IOException {
         ControllerVariables controllerVariables = webTargets.getControllerVariables();
-        updateStatus(ThingStatus.ONLINE);
+        var lastTransitionAgoSecs = Duration.between(lastTransition, Instant.now()).getSeconds();
+        var inTransition = lastTransitionAgoSecs < this.transitionTimeSecs;
         if (controllerVariables != null) {
+            updateStatus(ThingStatus.ONLINE);
             updateState(OpenGarageBindingConstants.CHANNEL_OG_DISTANCE,
                     new QuantityType<>(controllerVariables.dist, MetricPrefix.CENTI(SIUnits.METRE)));
-            boolean invert = isChannelInverted(OpenGarageBindingConstants.CHANNEL_OG_STATUS_SWITCH);
-            switch (controllerVariables.door) {
-                case 0:
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS, invert ? OnOffType.ON : OnOffType.OFF);
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS_SWITCH,
-                            invert ? OnOffType.ON : OnOffType.OFF);
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS_ROLLERSHUTTER, UpDownType.DOWN);
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS_CONTACT, OpenClosedType.CLOSED);
-                    break;
-                case 1:
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS, invert ? OnOffType.OFF : OnOffType.ON);
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS_SWITCH,
-                            invert ? OnOffType.OFF : OnOffType.ON);
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS_ROLLERSHUTTER, UpDownType.UP);
-                    updateState(OpenGarageBindingConstants.CHANNEL_OG_STATUS_CONTACT, OpenClosedType.OPEN);
-                    break;
-                default:
-                    logger.warn("Received unknown door value: {}", controllerVariables.door);
+            var maybeInvert = getInverter(OpenGarageBindingConstants.CHANNEL_OG_STATUS_SWITCH);
```

Comment:
I would prefer to avoid "var" in that place because with var you need to look at `getInverter` to guess the type.